packages/eas-cli/src/run/android/emulator.ts

```diff
@@ -0,0 +1,161 @@
+import spawnAsync, { SpawnResult } from '@expo/spawn-async';
+import chalk from 'chalk';
+import assert from 'node:assert';
+import os from 'os';
+
+import Log from '../../log';
+import { promptAsync } from '../../prompts';
+import {
+  AndroidEmulator,
+  adbAsync,
+  getFirstRunningEmulatorAsync,
+  isEmulatorBootedAsync,
+  waitForEmulatorToBeBootedAsync,
+} from './adb';
+import { sdkRoot } from './sdk';
+
+export const EMULATOR_MAX_WAIT_TIMEOUT = 60 * 1000 * 3;
+
+export const emulatorExecutable = getEmulatorExecutable();
+
+function getEmulatorExecutable(): string {
+  if (sdkRoot) {
+    return `${sdkRoot}/emulator/emulator`;
+  }
+
+  return 'emulator';
+}
+
+async function emulatorAsync(...options: string[]): Promise<SpawnResult> {
+  try {
+    return await spawnAsync(emulatorExecutable, options);
+  } catch (error: any) {
+    const errorMessage = (error.stderr || error.stdout || error.message).trim();
+    error.message = errorMessage;
+    throw error;
+  }
+}
+
+async function getAvaliableAndroidEmulatorsAsync(): Promise<AndroidEmulator[]> {
+  try {
+    const { stdout } = await emulatorAsync('-list-avds');
+
+    return stdout
+      .split(os.EOL)
+      .filter(Boolean)
+      .map(name => ({
+        name,
+      }));
+  } catch {
+    return [];
+  }
+}
+
+/** Start an Android device and wait until it is booted. */
+async function bootEmulatorAsync(
+  emulator: AndroidEmulator,
+  {
+    timeout = EMULATOR_MAX_WAIT_TIMEOUT,
+    interval = 1000,
+  }: {
+    /** Time in milliseconds to wait before asserting a timeout error. */
+    timeout?: number;
+    interval?: number;
+  } = {}
+): Promise<AndroidEmulator> {
+  Log.newLine();
+  Log.log(`Opening emulator ${chalk.bold(emulator.name)}`);
+
+  // Start a process to open an emulator
+  const emulatorProcess = spawnAsync(
+    emulatorExecutable,
+    [
+      `@${emulator.name}`,
+      // disable animation for faster boot -- this might make it harder to detect if it mounted properly tho
+      //'-no-boot-anim'
+    ],
+    {
+      stdio: 'ignore',
+      detached: true,
+    }
+  );
+
+  emulatorProcess.child.unref();
```

Comment:
From `child_process` JS Doc:
```
By default, the parent will wait for the detached child to exit. To prevent the parent from waiting for a given subprocess to exit, use the subprocess.unref() method. Doing so will cause the parent's event loop to not include the child in its reference count, allowing the parent to exit independently of the child unless there is an established IPC channel between the child and the parent.
```

In our case, we want the `eas build:run` command to finish once the app is installed and launched and we don't want to wait with it until the emulator is closed.

How would it look like without this line:

https://user-images.githubusercontent.com/55145344/198571339-6e3813b8-f387-4238-b105-f6d731be8a85.mov

I will also add a similar explanation in the code